pandas/tests/indexes/test_multi.py

```diff
@@ -1704,6 +1704,11 @@ def test_to_hierarchical(self):
         tm.assert_index_equal(result, expected)
```

Comment:
you need to catch all uses of to_hierarchical (or you get the warnings)

Reply:
Thanks - not sure if I understand. 
The only use of `to_hierarchical` was in `pandas/core/panel.py`- this has been substituted. 
This existing test is specifically for `to_hierarchical` - which can be removed once this method is removed in a future version - should these tests be removed as part of this PR?

Reply:
@jreback I think after reviewing changes in #18258 I probably get what is expected - please ignore the comment above - will update